waku/v2/utils/protobuf.nim

```diff
@@ -17,4 +17,10 @@ proc finish3*(proto: var ProtoBuffer) =
     proto.offset = 0
 
 proc `==`*(a: zint64, b: zint64): bool =
-  int64(a) == int64(b)
\ No newline at end of file
+  int64(a) == int64(b)
+
+proc uintToBool*(value: uint64): bool =
+  value != 0
+
+proc boolToUint*(value: bool): uint =
+  if value: 1 else: 0
```

Comment:
I don't understand why you need this 🤔 Protocol buffers supports the boolean type: https://developers.google.com/protocol-buffers/docs/proto3#:~:text=Int64-,bool,bool,-string

Reply:
While protobufs do support bools, our upstream implementation (minprotobuf) in nim-libp2p does not. Upon serialisation bools and uints are the same, hence the usage of uint :)

Reply:
@Menduist, does it make sense to do it this way, or would we rather create the bool field upstream?

Some additional usage of bool here - https://github.com/status-im/nwaku/blob/master/waku/v2/protocol/waku_filter/rpc_codec.nim#L45-L47

Reply:
As discussed, how we did it previously: https://github.com/status-im/nwaku/blob/master/waku/v2/protocol/waku_filter/rpc_codec.nim#L47

Reply:
You can open a PR to nim-libp2p